### DISCUSSION TITLE
How to track down a bug somewhere in vsomeip and CommonAPI?

### DISCUSSION THREAD

**Discussion by srl100** [COVESA/vsomeip#672] [Q&A] [answered]
I am trying to track down the bug responsible for this issue https://github.com/COVESA/capicxx-someip-runtime/issues/40 and I am wondering where to start.

The first step would seem to be to determine whether the problem is in vsomeip or CommonAPI, so are there any resources that let me split the problem out, e.g. tracing the messages between the two components?

**Comment by duartenfonseca** [comment, marked as answer]
hey there, i don't think there is a tool like that, however you could take a look at this CAPI examples: https://github.com/COVESA/capicxx-core-tools/blob/master/CommonAPI-Examples/E03Methods/README 
this one specifically implements a error code handling
